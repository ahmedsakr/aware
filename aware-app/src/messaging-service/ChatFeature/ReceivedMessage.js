import React, { Component } from 'react';
import './ReceivedMessage.css';

class ReceivedMessage extends Component {
    render() {
        const { student, avatar, message, timestamp } = this.props;

        return (
            <div class="receivedMessage">
                <div class="col-sm-2 receivedMessage-profile">
                    <img class="receivedMessage-profile-img" src={process.env.PUBLIC_URL + avatar} alt={student} />
                    <p class="receivedMessage-profile-name">{student}</p>
                </div>

                <div class="col-sm-7 receivedMessage-text">
                    {message}
                </div>

                <div class="col-sm-3 receivedMessage-timestamp">
                    {timestamp}
                </div>
            </div>
        );
    }
}

export default ReceivedMessage;
