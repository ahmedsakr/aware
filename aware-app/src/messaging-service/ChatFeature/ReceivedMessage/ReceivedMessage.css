.receivedMessage {
    margin-top: 10px;
    display: flex;
}

.receivedMessage-profile {
    border: 2px solid rgb(54, 56, 58);
    border-radius: 5px;
    text-align: center;
    padding: 0px;
    margin-right: 5px;
}

.receivedMessage-profile-img {
    max-width: 30%;
}

.receivedMessage-profile-name {
    background-color: #bac1b8;
    text-align: center;
    margin: 0px;
}

.receivedMessage-text {
    background-color: #a1c6ea;
    border-radius: 10px;
    padding: 5px;
    margin-right: 5px;
}

.receivedMessage-timestamp {
    padding: 0px;
    text-align: left;
    font-size: 0.6em;
}
