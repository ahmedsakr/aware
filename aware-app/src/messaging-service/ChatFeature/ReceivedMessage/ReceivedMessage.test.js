import React from 'react';
import { shallow } from 'enzyme';

import ReceivedMessage from './ReceivedMessage';

describe('ReceivedMessage', () => {
  it('should render correctly in "debug" mode', () => {

    const component = shallow(<ReceivedMessage debug />);
  
    expect(component).toMatchSnapshot();
  });
});