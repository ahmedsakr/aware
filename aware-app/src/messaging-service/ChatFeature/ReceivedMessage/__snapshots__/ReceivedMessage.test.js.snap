// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReceivedMessage should render correctly in "debug" mode 1`] = `
<div
  class="receivedMessage"
>
  <div
    class="col-sm-2 receivedMessage-profile"
  >
    <img
      class="receivedMessage-profile-img"
      src="undefined"
    />
    <p
      class="receivedMessage-profile-name"
    />
  </div>
  <div
    class="col-sm-7 receivedMessage-text"
  />
  <div
    class="col-sm-3 receivedMessage-timestamp"
  />
</div>
`;
